x-pack/plugins/fleet/public/applications/fleet/hooks/use_apm_service_href.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Overwrite } from 'utility-types';
+import useAsync from 'react-use/lib/useAsync';
+
+import { useLocator } from '../../../hooks/use_locator';
+import type { DataStream } from '../types';
+import { LOCATORS_IDS } from '../constants';
+
+type APMDataStream = Overwrite<DataStream, { package: 'apm' }>;
+
+const isAPMIntegration = (datastream: DataStream): datastream is APMDataStream =>
+  Boolean(datastream.package === 'apm');
+
+const hasServiceDetails = (
+  datastream: DataStream
+): datastream is Overwrite<
+  DataStream,
+  { package: 'apm'; serviceDetails: NonNullable<DataStream['serviceDetails']> }
+> => Boolean(datastream.serviceDetails);
+
+export const useAPMServiceDetailHref = (datastream: DataStream) => {
+  const apmLocator = useLocator(LOCATORS_IDS.APM_LOCATOR);
+
+  const { error, loading, value } = useAsync(() => {
+    if (!isAPMIntegration(datastream) || !apmLocator) return Promise.resolve();
+
+    if (hasServiceDetails(datastream)) {
+      const { serviceName, environment } = datastream.serviceDetails;
+
+      return apmLocator.getUrl({
+        serviceName,
+        serviceOverviewTab: datastream.isErrorOnlyLogsStream ? 'errors' : datastream.type,
```

Comment:
I think we should redirect to the errors tab here if type == 'logs', given that we'll only write errors to logs (AFAIK).